src/ascii/port.rs

```diff
@@ -116,10 +89,18 @@ pub struct Port<'a, B> {
     poison: Option<io::Error>,
     /// The builder used to concatenate packets in to responses.
     builder: ResponseBuilder,
-    /// Optional hook to call after a packet is sent/received.
-    packet_hook: Option<PacketCallbackDebugWrapper<'a>>,
-    /// Optional hook to call when an unexpected Alert is received.
-    unexpected_alert_hook: Option<UnexpectedAlertDebugWrapper<'a>>,
+    /// User supplied callbacks
+    callbacks: C,
+    /// Marker for the lifetime
+    lifetime: std::marker::PhantomData<&'a std::convert::Infallible>,
+}
+
+impl<'a, B: std::fmt::Debug, C> std::fmt::Debug for Port<'a, B, C> {
```

Comment:
`Box<dyn Backend>` doesn't implement `Debug`, which is easily created by users via the `open_dyn` methods. All backends users can user implement `Backend`, so that would likely be a better bound.

Reply:
be0534f